### ISSUE TITLE
Footstep sound max hear range of mobs is too high

### ISSUE THREAD

**Issue by MarkuBu** [minetest/minetest#4361]
I start a short test game with the current version of mobs redo and increased values for max_block_generate_distance and max_block_send_distance

The problem is, that I can hear footstep sounds from everywhere, even if there is no other cave near. I guess, I can hear them in about 60-70 nodes, maybe more. But I would say that 10 nodes would be enough, maybe 15 nodes maximum

You can hear them everywhere and become crazy

**Comment by paramat** [comment]
16 would seem enough to me +1.

**Comment by MarkuBu** [comment]
How can this be changed? Maybe, I can make a PR if I know in which file.

**Comment by SmallJoker** [comment]
Here it is: https://github.com/minetest/minetest/blob/master/src/content_cao.cpp#L1216
According to the file `sound.h` the function call would look like this:
```c++
m_gamedef->sound()->playSoundAt(spec, false, VOLUME_HERE, getPosition());
```

**Comment by MarkuBu** [comment]
Ok then someone else must do this. I don't have enough experience with C++ and the Minetest code

But this should be really fixed because it is annoying and useless

**Comment by paramat** [comment]
See closed duplicate #6131

**Comment by Wuzzy2** [comment]
Relevant quote:

> If I am not mistaken, the sound also has 100% volume for the entire hear distance which is very bad.
> A mob which is 31 nodes away is as loud as a mob which is only 1 node away. And a mob which is 33 nodes away is silent.

(correct me if I'm wrong)

**Comment by paramat** [comment]
Yes if true, as stated in the other thread, that needs fixing, or if it has to be fixed, perhaps a lower level for other players?

**Comment by paramat** [comment]
Relevant code is here https://github.com/minetest/minetest/blob/master/src/content_cao.cpp#L865
```
			if(!m_is_local_player && m_prop.makes_footstep_sound)
			{
				INodeDefManager *ndef = m_client->ndef();
				v3s16 p = floatToInt(getPosition() + v3f(0,
						(m_prop.collisionbox.MinEdge.Y-0.5)*BS, 0), BS);
				MapNode n = m_env->getMap().getNodeNoEx(p);
				SimpleSoundSpec spec = ndef->get(n).sound_footstep;
				m_client->sound()->playSoundAt(spec, false, getPosition());
}
```

**Comment by paramat** [comment]
Here's a guess: The sound is attenuating but the gain is set too high somehow, and the volume is therefore hitting a maximum limit and removing the fade with distance, how about we try adding a lower gain to see if this is the case?

**Comment by paramat** [comment]
![screenshot_20170919_160005](https://user-images.githubusercontent.com/3686677/30599283-e77c1b2e-9d53-11e7-92f6-ec78f011a8a9.png)

Tested with a NPC from mobs redo trapped in a glass corridor, and marked a line 30 nodes away, 30 being the default max hear distance in openAL.
Footstep sounds do attenuate with distance and become inaudible at 30 nodes.
I have noticed that the mob footsteps can be louder than the player's own when the mob is close, maybe we can reduce mob footstep volume a little to compensate.
I agree 30 nodes is a little too far, 16 seems good, if i can i will reduce the max range.

**Comment by paramat** [comment]
PR #6440 
Needs testing on a multiplayer server.

**Comment by paramat** [comment]
#6440 merged which will help reduce hear distance.
Changing thread title to include #6131

**Comment by paramat** [comment]
> The problem is, that I can hear footstep sounds from everywhere, even if there is no other cave near. I guess, I can hear them in about 60-70 nodes, maybe more.

My tests do not support this, if this happens it must be a bug.

At the least i think we can reduce the distance to 16 nodes or less, hearing footsteps from 30 nodes is ridiculous. This i support.
A setting i am not sure, footsteps are something that makes sense to have a fixed max hear distance.
